### PR TITLE
feat: don't send event invites if event starts in the past

### LINKED ISSUE #1661: Members of the chapter shouldn't get notified by the creation of an event in the past
### Describe the feature:

As discussed in Discord, owner may want to create events for documentation purposes, but creating this events may result in notification been sent to the chapter members. 

We need some flag which stops the events, that were created before (`new Date()`), from using our notification system

### PR DESCRIPTION
**Pull request by gikf**
- [x] I have read [Chapter's contributing guidelines](https://github.com/freeCodeCamp/chapter/blob/main/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update README.md`).
- [x] My pull request targets the `main` branch of Chapter.

Closes #1661

---

- Doesn't send event invite if event starts in the past. I think it's not needed to be very precise with that check.
- Changes default event start and end by one day.
- Wording for note and positioning could be better.

### CODE REVIEW

**Comment by gikf** [comment]
I don't know what's the issue with test and I'm unable to reproduce it locally :thinking:

**Comment by gikf** [comment]
Changing the system time first appears to make test pass. I'm not very convinced that was the main issue, it looks more to me like some fix as a side-effect.

**Comment by Sboonny** [comment]
Is the issue in `eventWithoutUrl`? as it start in the past, but the new function doesn't send email about old events.

```json
 "eventWithoutURL": {
    "name": "Test Event",
    "description": "Test Description",
    "streaming_url": "https://test.event.org/video",
    "capacity": "10",
    "start_at": "2022-01-01T00:01",
    "ends_at": "2022-01-02T00:02",
    "venue_id": "1",
    "image_url": "https://test.event.org/image"
  },
 ```

**Comment by gikf** [comment]
That default date should be overwritten in the `testEvent` here:
https://github.com/freeCodeCamp/chapter/blob/0280478fa16418fc89644c83cc994d8b13978af0/cypress/e2e/dashboard/chapters/%5BchapterId%5D/chapterId-index.cy.ts#L88-L92

**Review by Sboonny — APPROVED** [review]
LGTM 🎉, thank you for your contribution to the page! 👍.

**Comment by Sboonny** [comment]
> That default date should be overwritten in the `testEvent` here:
> 
> https://github.com/freeCodeCamp/chapter/blob/0280478fa16418fc89644c83cc994d8b13978af0/cypress/e2e/dashboard/chapters/%5BchapterId%5D/chapterId-index.cy.ts#L88-L92

Yeah true, I can't think of why 😞. Should we add a comment about the issue in cypress file for context?

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]
@gikf nice work as always.

It's possible it could be organised a little differently, though.  Let me know what you think.

File: client/src/modules/dashboard/Events/components/EventDatesForm.tsx [hunk 1]
```diff
@@ -55,48 +97,40 @@ const EventDatesForm: React.FC<EventDatesFormProps> = ({
...
-                value={date.toDateString()}
-              />
-            }
-          />
-        </FormControl>
-      ))}
+      <FormControl isRequired={startDateProps.isRequired}>
+        <>
+          {datePicker(startDateProps)}
+          {!startAt && isPast(startDate) && (
```
[the remark this answers is not in the document]
Comment (gikf) [reply client/src/modules/dashboard/Events/components/EventDatesForm.tsx:120]: Yeah, it is. I'm gonna do that. I thought about some explicit prop for that, but couldn't really justify it to myself at the time.

File: client/src/modules/dashboard/Events/components/EventDatesForm.tsx [hunk 2]
```diff
@@ -1,16 +1,18 @@
...
 import { useFormContext } from 'react-hook-form';
+import { add, isPast } from 'date-fns';
 
 import 'react-datepicker/dist/react-datepicker.css';
 
 import { Input } from '../../../../components/Form/Input';
 
 interface EventDatesFormProps {
-  endsAt: Date;
+  endsAt: Date | null;
```
[the remark this answers is not in the document]
Comment (gikf) [reply client/src/modules/dashboard/Events/components/EventDatesForm.tsx:13]: You are right :+1:.

File: client/src/modules/dashboard/Events/components/EventDatesForm.tsx [hunk 3]
```diff
@@ -41,6 +49,40 @@ const EventDatesForm: React.FC<EventDatesFormProps> = ({
...
+          name={key}
+          error={error}
+          label={`${label}${isRequired ? ' (Required)' : ''}`}
+          isDisabled={loading}
+          isRequired={isRequired}
+          value={date.toDateString()}
+        />
+      }
+    />
+  );
```
[the remark this answers is not in the document]
Comment (ojeytonwilliams) [reply client/src/modules/dashboard/Events/components/EventDatesForm.tsx:84]: I'm probably a little too conservative when it comes to stuff like this.  This isn't a deeply weird pattern I've never seen before (we use it the fcc repo after all).

**Review by ojeytonwilliams — APPROVED** [review]

**Pull request merged**